### ISSUE TITLE
Material logger only take logs at ship destruction.

### ISSUE THREAD

**Issue by icylogic** [yuyuvn/KanColleViewer#64]
It seems that in these lines, material logger subcribes 3 events: api_get_member_material, api_req_hokyu_charge and api_req_kousyou_destroyship
https://github.com/yuyuvn/KanColleViewer/blob/master/Grabacr07.KanColleWrapper/Logger.cs#L83-85
But in ver.3.8.2r556, it only take logs at ship destruction.

In addition, the counter of "resupply times" works properly

**Comment by gakada** [comment]
Are you sure?

https://github.com/yuyuvn/KanColleViewer/blob/v3.8.2r556/Grabacr07.KanColleWrapper/Logger.cs#L83-85

I downloaded and tried in ver.3.8.2r556 and it logs in MaterialsLog.csv on resupply (kcsapi_charge).

**Comment by gakada** [comment]
So `kcsapi_charge` and `kcsapi_destroyship` work, while `kcsapi_material` isn't (?), I guess it supposed to be called on ship construction, but

https://github.com/yuyuvn/KanColleViewer/blob/master/Grabacr07.KanColleWrapper/Logger.cs#L187

while I get source.Length == 8, so maybe this should be changed.

Also, we can subscribe to `createitem`.

Or not subscribe to any of these and subscribe to `port` instead, while tracking resource changes in Logger, that way it will log every time you go to the homeport and when resources had changed.

**Comment by icylogic** [comment]
@gakada 

Thanks for your quick response.

Today I've resupplied 24 times.
![2015-07-30_19-21-03](https://cloud.githubusercontent.com/assets/2814357/8982077/390b851e-36f0-11e5-985c-267ded72ed57.png)
And there are only 3 material logs which immediately occurred as I destroyed ships.
![2015-07-30_19-20-11](https://cloud.githubusercontent.com/assets/2814357/8982064/1d5b7694-36f0-11e5-867c-8a3f14cc60fa.png)

Is it possible that the new type of material ("改修", equipment improving) enlarge the materials list, so the length changes to 8? I've not check it yet. At least these usages of magic number is not a good style.

I agree that we could simply subscribe to port, since almost all events happen with changes of materials in fact.
